walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,65 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
 
 public final class FindMeetingQuery {
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<Event> sortedEvents = new ArrayList<Event>(events);
+    Collection<TimeRange> freeTimeRanges = new ArrayList<TimeRange>();
+    Collection<TimeRange> freeTimeRangesWithOptionalAttendee = new ArrayList<TimeRange>();
+    int end = 0, FULL_DAY = 24*60, endMandatory = 0;
+    
+    Collections.sort(sortedEvents, new Comparator<Event>() {
+      @Override public int compare(Event e1, Event e2) {
+        return TimeRange.ORDER_BY_START.compare(e1.getWhen(), e2.getWhen());
+      }
+    });
+    
+    for(Event e: sortedEvents) {
+      if(!Collections.disjoint(e.getAttendees(), request.getAttendees())) {
```

Comment:
Style nit: prefer
`if (cond) { continue; } doTheRest();`
over
`if (!cond) ( doTheRest(); )`
Chances are doTheRest will have more nested calls and there will be too many layers if it already started within a if block.